### PR TITLE
Update examples how to use VersionedLayerClient and VolatileLayerClient

### PR DESCRIPTION
**Pull request by AndriyShevchuk**
* Update example how to use VersionedLayerClient
* Update example how to use VolatileLayerClient

Resolves: OLPEDGE-865
Signed-off-by: ashevchu <ext-andrii.shevchuk@here.com>

### CODE REVIEW

File: docs/examples/nodejs-read-versioned-layer.md [hunk 1]
```diff
@@ -51,22 +51,23 @@ class App {
...
-```
+```shell
 App works!
 ```
 
 ## Log into the data store
 
-The first thing you need is https://account.here.com account.
-Log into your account and create a test app to get app access key id and secret access key at https://platform.here.com/admin/apps
+The first thing you need is [account](https://account.here.com).[link](https://google.com)
```
Comment (OleksiiZubko) [inline docs/examples/nodejs-read-versioned-layer.md]: Please fix google

File: docs/examples/nodejs-read-versioned-layer.md [hunk 2]
```diff
@@ -76,39 +77,41 @@ Log into the api. Modify our app to the following state:
...
 });
 
 ```
 
-## Context
+## OlpClientSettings
 
 Now you can initialize DataStore client for `here` environment and, for example, list catalogs.
-To do so you need a `DatastoreContext`. It contains look up api functionallity to get the base urls to the services.
+To do so you need a `OlpClientSettings`. It contains look up api functionallity to get the base urls to the services.
```
Comment (OleksiiZubko) [inline docs/examples/nodejs-read-versioned-layer.md]: OlpClientSettings does not contain lookup api func... . It contains download manager, token provider, and key-value cache

File: docs/examples/nodejs-read-versioned-layer.md [hunk 3]
```diff
@@ -117,73 +120,73 @@ const context = new DataStoreContext({
...
-
+To create `VersionedLayerClient`, run:
 
 ```typescript
-import { UserAuth } from "@here/olp-sdk-authentification";
-import { DataStoreContext, VersionedLayerClient } from "@here/olp-sdk-dataservice-read";
-import * as utils from "@here/olp-sdk-dataservice-read";
+//  Client for "protobuf-example-berlin-v1" layer from "sensor-data-sensoris-versioned-example" catalog
+const versionClient = await new VersionedLayerClient(
+    HRN.fromString("hrn:here:data:::sensor-data-sensoris-versioned-example"),
```
Comment (OleksiiZubko) [inline docs/examples/nodejs-read-versioned-layer.md]: please, don't use the real hrn, datahandles, etc ..

File: docs/examples/nodejs-read-versioned-layer.md (hunk 1 above)
Comment (AndriyShevchuk) [reply docs/examples/nodejs-read-versioned-layer.md]: Removed

File: docs/examples/nodejs-read-versioned-layer.md (hunk 2 above)
Comment (AndriyShevchuk) [reply docs/examples/nodejs-read-versioned-layer.md]: Fixed

File: docs/examples/nodejs-read-versioned-layer.md (hunk 3 above)
Comment (AndriyShevchuk) [reply docs/examples/nodejs-read-versioned-layer.md]: Removed

File: docs/examples/nodejs-read-versioned-layer.md [hunk 4]
```diff
@@ -1,35 +1,34 @@
...
-This example shows how to retrieve partition metadata, and partition data using the OLP SDK for TypeScript.
+This example shows how to retrieve partition metadata and partition data using the OLP SDK for TypeScript.
 
 ## Build and running on Node.js
 
 Requirements:
 
-Node.js 10+
-TypeScript: 3.5+
+See [README.md](../../README.md#Dependencies) for dependencies
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
For information on dependencies, see [README.md](../../README.md#Dependencies).
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 5]
```diff
@@ -1,35 +1,34 @@
...
 
-Start by creating a simple Node.js app:
+Create a simple Node.js app:
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
Create the Node.js app:
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 6]
```diff
@@ -1,35 +1,34 @@
...
 ```
 
 Then initialize a TypeScript project:
 
-```
+```shell
 tsc --init
 ```
 
 Also, install node types:
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
Install node types:
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 7]
```diff
@@ -1,35 +1,34 @@
...
 
-```
+```shell
 npm install --save-dev @types/node
 ```
 
 Then install SDK modules:
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
Install SDK modules:
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 8]
```diff
@@ -51,22 +50,23 @@ class App {
...
 ```
 
 After a successful run, the console displays the following message:
 
-```
+```shell
 App works!
 ```
 
 ## Log into the data store
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
## Log in to the datastore
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 9]
```diff
@@ -51,22 +50,23 @@ class App {
...
 
-The first thing you need is https://account.here.com account.
-Log into your account and create a test app to get app access key id and secret access key at https://platform.here.com/admin/apps
+The first thing you need is [account](https://account.here.com).
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
To work with the datastore, you need to have an [account](https://account.here.com).
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 10]
```diff
@@ -51,22 +50,23 @@ class App {
...
+Log into your account and create a test app to get app access key id and secret access key at [platform.here.com](https://platform.here.com/admin/apps)
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
To get your app access key id and access key secret, log in to the [HERE Open Location Platform (OLP)](https://platform.here.com/admin/apps) and create a test app.
```

File: docs/examples/nodejs-read-volatile-layer.md [hunk 11]
```diff
@@ -50,64 +50,67 @@ class App {
...
 ```
 
 After a successful run, the console displays the following message:
 
-```
+```shell
 App works!
 ```
 
 ## Log in to the data store
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-volatile-layer.md]:
```suggestion
## Log in to the datastore
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 12]
```diff
@@ -51,22 +50,23 @@ class App {
...
 
 Now you can obtain the token for requests to the datastore by using the @here/olp-sdk-authentification.
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
Now, you can get a token for requests to the datastore using the @here/olp-sdk-authentification.
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 13]
```diff
@@ -76,39 +76,41 @@ Log into the api. Modify our app to the following state:
 /**
  * Example Node.js app for reading versioned layer from datastore
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
 * Example Node.js app for reading a versioned layer from the datastore
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 14]
```diff
@@ -76,39 +76,41 @@ Log into the api. Modify our app to the following state:
...
+    },
+    tokenRequester: requestToken
 });
 
 ```
 
-## Context
+## OlpClientSettings
 
 Now you can initialize DataStore client for `here` environment and, for example, list catalogs.
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
Now, you can initialize the `DataStore` client for the `here` environment and, for example, list catalogs.
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 15]
```diff
@@ -76,39 +76,41 @@ Log into the api. Modify our app to the following state:
...
-To do so you need a `DatastoreContext`. It contains look up api functionallity to get the base urls to the services.
+To do so you need a `OlpClientSettings`. It contains download manager, token provider, and key-value cache.
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
To do so, you need the `OlpClientSettings` object. It contains a download manager, token provider, and key-value cache.
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 16]
```diff
@@ -76,39 +76,41 @@ Log into the api. Modify our app to the following state:
...
 
 ```typescript
 /**
  * Example Node.js app for reading versioned layer from datastore
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
 * Example Node.js app for reading a versioned layer from the datastore.
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 17]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
 
 ## VersionedLayerClient
 
-Therefore, when you have context, you can get catalog clients for different catalogs and read the information.
+Therefore, when you have settings, you can get catalog clients for different catalogs and read the information.
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
When you have the `OlpClientSettings` object, you can get catalog clients for different catalogs and read information from catalogs.
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 18]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
 You can get more information about Version Layer following the [link](https://developer.here.com/olp/documentation/data-api/data_dev_guide/rest/publishing-data-versioned.html).
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
For information on a versioned layer following, see [the related section](https://developer.here.com/olp/documentation/get-started/dev_guide/shared_content/topics/olp/concepts/layers.html#versioned-layers) in the Get Started guide.
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 19]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
-To create VersionedLayerClient, run:
-
+To create `VersionedLayerClient`, run:
 
 ```typescript
-import { UserAuth } from "@here/olp-sdk-authentification";
-import { DataStoreContext, VersionedLayerClient } from "@here/olp-sdk-dataservice-read";
-import * as utils from "@here/olp-sdk-dataservice-read";
+//  Client for "protobuf-example-berlin-v1" layer from "sensor-data-sensoris-versioned-example" catalog
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
//  Client for the "protobuf-example-berlin-v1" layer from the "sensor-data-sensoris-versioned-example" catalog.
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 20]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
+2. getPartitions() - to fetch partitions metadata
 
-//  Client for "protobuf-example-berlin-v1" layer from "sensor-data-sensoris-versioned-example" catalog
-const versionClient = await new VersionedLayerClient({
-    context,
-    hrn: "hrn:here:data:::sensor-data-sensoris-versioned-example",
-    layerId: "protobuf-example-berlin-v1",
-    version: 0
-});
+Method `getData()` expects 2 arguments:
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
The `getData()`  method expects the following arguments:
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 21]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
 
-// Get partition "23618173" by id
-const partition = await versionClient.getPartition('23618173');
+* dataRequest - `DataRequest` instanse. Class that prepare data for the requests to the BlobAPI. `getData` method can fetch partition data by 3 types of parameters you can provide to it. It could be next parameters sorted by priority: `dataHandle`, `partitionId` and `quadKey`. Below you can take a look on the example of creating `DataRequest` instanse.
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
* `dataRequest` &ndash;  the `DataRequest` instanse. This class prepares data for the requests to the BlobAPI. The `getData` method can fetch partition data by the following 3 types of parameters (sorted by priority): `dataHandle`, `partitionId`, and `quadKey`. Below, find the example of how to create `DataRequest` instanse.
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 22]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
+* abortSignal - a signal object that allows you to communicate with a request (such as a Fetch) and abort it if required via an AbortController object. More details see [here](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal).
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
* `abortSignal` &ndash; a signal object that allows you to communicate with the request (such as `fetch`) and, if required, abort itusing the `AbortController` object. For more information, see the [`AbortSignal` documentation](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal).
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 23]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
-const tile = await versionClient.getTile({
-    row: 1,
-    column: 2,
-    level: 3
-});
+```typescript
+import { DataRequest } from "@here/olp-sdk-dataservice-read";
+// DataRequest usage example
+const request = new DataRequest();
+// add `dataHandle` property
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
// Add the `dataHandle` property.
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 24]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
+request.withDataHandle("TEST24A111D82321A9BA9071A7EF042.042");
+// add `PartitionId` property
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
// Add the `PartitionId` property.
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 25]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
+request.withPartitionId("123123123");
+// add `QuadKey` property
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
// Add the `QuadKey` property.
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 26]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
+request.withQuadKey(quadKeyFromMortonCode("123121122"));
 
-// Get tile using mortonCode
-const tile = await versionClient.getTile(utils.quadKeyFromMortonCode("1476147"));
+//Also data could be added by chain like so
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
//Also, data can be added by the following chain:
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 27]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
+const requestByChain = new DataRequest().withPartitionId("123123123").withQuadKey(quadKeyFromMortonCode("123121122"));
+```
 
-// Get aggregated tile using mortonCode
-const aggregatedTile = await versionClient.getAggregatedTile(utils.quadKeyFromMortonCode("1476147"));
+Now let's get data
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
Now, you can get data from the layer.
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 28]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
-// Get all partitions metadata for this layer
-const partitions = await versionClient.getPartitionsMetadata();
+```typescript
+const request = new DataRequest().withDataHandle("TESTE24A111D82321A9BA9071A7EF042.042");
+const result = await versionClient.getData(request);
+```
 
-// Get data coverage bitmap for this layer
-const bitmap = await versionClient.getDataCoverageBitMap();
+Method `getPartitions` expects 2 arguments:
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
The `getPartitions` method expects the following arguments:
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 29]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
 
-// Get data coverage sizemap for this layer
-const sizeMap = await versionClient.getDataCoverageSizeMap();
+* QuadKeyPartitionsRequest `or` PartitionsRequest`.
+* abortSignal - a signal object that allows you to communicate with a request (such as a Fetch) and abort it if required via an AbortController object. More details see [here](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal).
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
* `abortSignal` &ndash; a signal object that allows you to communicate with the request (such as `fetch`) and, if required, abort itusing the `AbortController` object. For more information, see the [`AbortSignal` documentation](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal).
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 30]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
 
-// Get data coverage timemap for this layer
-const timeMap = await versionClient.getDataCoverageTimeMap();
+To fetch partitions metadata from Query API by `QuadKey` use `QuadKeyPartitionsRequest`. It expects quadKey, depth (from 0 to 4) and version (optional).
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
To fetch partitions metadata from Query API by `QuadKey`, use `QuadKeyPartitionsRequest`. It expects a quadkey, depth (from 0 to 4), and version (optional).
```

File: docs/examples/nodejs-read-versioned-layer.md [hunk 31]
```diff
@@ -117,73 +119,73 @@ const context = new DataStoreContext({
...
+const request = new QuadKeyPartitionsRequest()
+                    .withQuadKey(quadKeyFromMortonCode("75717"))
+                    .withDepth(3) // default value is 0
+                    .withVersion(42); // optional
+const result = await versionClient.getPartitions(request);
+```
 
-// Get layer summary metadata
-const summary = await versionClient.getSummary();
+To fetch partitions metadata from MetadataAPI use `PartitionsRequest`. It expects version parameter (optional). If you skip version and pass empty PartitionsRequest instanse, than the last layer verion will be used.
```
Comment (HalynaDumych) [inline docs/examples/nodejs-read-versioned-layer.md]:
```suggestion
To fetch partitions metadata from the MetadataAPI, use `PartitionsRequest`. It expects version parameter (optional). If you skip version and pass an empty `PartitionsRequest` instanse, the last layer version will be used.
```

**Pull request merged**